Source/Core/InputCommon/GCAdapter.cpp

```diff
@@ -37,6 +37,7 @@ static std::atomic<int> s_controller_payload_size = {0};
 static std::thread s_adapter_thread;
 static Common::Flag s_adapter_thread_running;
 
+static std::recursive_mutex s_init_mutex;
```

Comment:
I’m using it because ResetRumble() is called from AddGCAdapter() (which is already locked) but also from other threads (which are not).